src/python/pants/base/exiter.py

```diff
@@ -51,11 +53,10 @@ def exit(self, result=0, msg=None, out=None):
     :param out: The file descriptor to emit `msg` to. (Optional)
     """
     if msg:
-      out = out or sys.stderr
-      # sys.stderr expects bytes in Py2, unicode in Py3
+      out = out or (sys.stderr.buffer if PY3 else sys.stderr)
       msg = ensure_binary(msg)
       try:
-        print(msg, file=out)
+        out.write(msg + b'\n')
```

Comment:
I imagine `msg` is usually short, but I abhor unnecessary string concatenation. Maybe instead:
 
```
out.write(msg)
out.write(b'\n')
```

?